modules/decklinksrc/decklinksrc.py

```diff
@@ -28,12 +31,23 @@ class DeckLinkSRC:
         Displays live video feed of DeckLink stream
     """
 
-    def __init__(self):
+    def __init__(self, videoPipeline):
         """
         Initializes DeckLink video stream
         """
         self.__currentFrame = None
-        self.capture = cv2.VideoCapture(0) #Starts capture on initialization of object
+        self.capture = cv2.VideoCapture(0)  # Starts capture on initialization of object
+        self.videoPipeline = videoPipeline
+        mainProcess = mp.Process(target=self._mainProcess_, daemon=True)
+        mainProcess.start()
+
+    def _mainProcess_(self):
+        print("decklink started")
+        while True:
+            newFrame = self.grab()
```

Comment:
Is this guaranteed to return a valid frame? We do not want invalid data in the pipeline between processes. This is applicable to all pipelines.